sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/SentrySpringFilter.java

```diff
@@ -60,7 +61,7 @@ protected void doFilterInternal(
     if (scopes.isEnabled()) {
       // request may qualify for caching request body, if so resolve cached request
       final HttpServletRequest request = resolveHttpServletRequest(servletRequest);
-      scopes.pushScope();
+      final @NotNull ISentryLifecycleToken lifecycleToken = scopes.pushIsolationScope();
```

Comment:
Is there a reason you're not using `scopes.withScope {}` instead here?

Reply:
We could create `withIsolationScope` and use that.

We can also use `try(... pushIsolationScope) { ... }` and it should do pretty much the same thing as `withScope` except it doesn't pass in the scope as a param.

Reply:
I would opt for doing try-with-resource whenever possible, if that makes sense to you